### PR TITLE
Introduce async/await style tests

### PR DESCRIPTION
**Pull request by opsb**

### CODE REVIEW

**Comment by dgeb** [comment]
Seems good except for having to disable hinting.

**Comment by dgeb** [comment]
@opsb if you're fine with #108, please merge and then re-enable hinting in this PR. I think a rebase may be needed as well.

File: tests/integration/store-test.js [hunk 1]
```diff
@@ -16,118 +16,66 @@ module('Integration - Store', function(hooks) {
...
-      .catch(error => {
-        assert.ok(error.message.match(/Record not found/));
-      });
-  });
-
-  test('#query - record', function(assert) {
-    let earth;
+  test('#removeRecord', async assert => {
+    const record = await store.addRecord({ type: 'planet', name: 'Earth' });
+    store.removeRecord(record);
```
Comment (dgeb) [inline tests/integration/store-test.js]: I believe this needs an `await`

Comment (opsb) [inline tests/integration/store-test.js]: yup, good catch

**Comment by dgeb** [comment]
Thanks @opsb

**Pull request merged**